### ISSUE TITLE
Inject grub options

### ISSUE THREAD

**Issue by mudler** [kairos-io/kairos#398]
**Is your feature request related to a problem? Please describe.**
osbuilder allows to build raw disks, however there is no way to setup automatic grub options ot the system

**Describe the solution you'd like**
The API to expose a set of grub options that get applied on the disk

**Describe alternatives you've considered**
N/A

**Additional context**

**Comment by ci-robbot** [comment]
Triage: this is implemented, and I think it can close. There is one trap worth writing down before it does, because which key you use depends on how the disk was produced.

## The API exists, in two places

| Key | Type | Declared | JSON schema |
|---|---|---|---|
| `grub_options` (top level) | `map[string]string` | `sdk/types/config/config.go:32` | `sdk/schema/root_schema.go:20` |
| `install.grub_options` | `map[string]string` | `sdk/types/install/install.go:21` | `sdk/schema/install_schema.go:18` |

Both feed `grubOptions()` in `agent/internal/agent/hooks/gruboptions.go`, which writes them into `grubenv` with `grub2-editenv` (OEM partition normally, STATE when OEM is encrypted). `extra_cmdline` is the interesting one for this issue: it is what ends up on the kernel command line.

## Which one applies to an osbuilder/AuroraBoot raw disk

This is the part that is not obvious:

- **`install.grub_options`** is consumed by `GrubPostInstallOptions` (`gruboptions.go:20`), which only runs as part of the agent's install flow.
- **top-level `grub_options`** is consumed by `GrubFirstBootOptions` (`gruboptions.go:80`), which is in the `FirstBoot` hook list (`agent/internal/agent/hooks/hook.go:34-38`) and fires from `agent/internal/agent/agent.go:42-52`, gated on the `/usr/local/.kairos/sentinel_firstboot` sentinel.

A raw disk is pre-installed, so the install flow never runs on it. AuroraBoot's raw image builder only drops the cloud config onto the OEM partition as `90_custom.yaml` (`pkg/ops/rawDiskGeneration.go:114-121`) and never invokes the agent's install hooks. So for exactly the case this issue describes, **the top-level `grub_options` key is the one that works**, and `install.grub_options` is silently ignored.

That is the feature the issue asked for, so: closing as implemented, with the caveat that the docs should be explicit about which key applies to prebuilt raw disks. Happy to open a docs PR for that if it is wanted.

## One small thing I noticed while checking

`GrubPostInstallOptions` short-circuits on `utils.IsUki()` (`gruboptions.go:21-24`), because `grubenv` is meaningless under UKI. `GrubFirstBootOptions` has no such guard, so on a UKI raw disk with top-level `grub_options` set it will mount OEM and write a `grubenv` nothing reads. Harmless, but dead work and a confusing log line. Not worth its own issue unless someone wants it; say the word and I will fold the guard in.
